src/vs/platform/actionWidget/browser/actionList.ts

```diff
@@ -24,6 +24,7 @@ export const previewSelectedActionCommand = 'previewSelectedCodeAction';
 export interface IActionListDelegate<T> {
 	onHide(didCancel?: boolean): void;
 	onSelect(action: T, preview?: boolean): void;
+	onFocus(action: T): Promise<boolean> | undefined;
```

Comment:
It's not clear what the return here does, jsdoc would be nice on the interface as a minimum but it seems a little weird for a focus event to return a boolean. Should this be a named object instead? `Promise<{ theThing: boolean }>`?

Also should onFocus be optional as the terminal doesn't need to implement it?